whisper/shhapi/api.go

```diff
@@ -368,60 +368,21 @@ func (api *PublicWhisperAPI) Post(args PostArgs) error {
 
 type PostArgs struct {
 	TTL      uint32              `json:"ttl"`
-	From     rpc.HexBytes        `json:"from"`
-	To       rpc.HexBytes        `json:"to"`
+	From     string              `json:"from"`
+	To       string              `json:"to"`
```

Comment:
I did so because it was more convenient to do it this way. This type (PostArgs) is only used once in a single function PublicWhisperAPI.Post. Inside this function all the necessary checks and validations are performed. Not only the one you have mentioned, but many others. Invalid parameters will be rejected. Also, I've written a lot of tedious test routines to check the parameters, their types, interdependency, etc.